### ISSUE TITLE
Using with importmaps

### ISSUE THREAD

**Issue by crespire** [stimulus-components/stimulus-carousel#11]
Anyone have any tips on how to get this running using import maps? I've pinned the Swiper package, and I've imported the CSS via `@import url` in my `application.scss` but no dice.

**Comment by crespire** [comment]
Looks like the implementation on the site was somewhat incompatible with importmaps. I was able to get everything up and running like so:

1. Pin swiper via `./bin/importmap pin swiper`
2. Add `import "swiper"` to `application.js`
3. Import the CSS via `@import url('https://cdn.jsdelivr.net/npm/swiper@9.3.2/swiper-bundle.min.css')` in your `application.scss` - note this version number is simply the current version as of this comment.
4. Get the Stimulus controller code into your application. I've been copy/pasting as we are not using yarn and also not using TS, so I edit to down to regular JS.

The import in the Stimulus controller is a bit different than documented:
`import Swiper, { Navigation, Pagination } from 'swiper'`

Then rather than use a method for the default options, I simply init Swiper per their API docs in the `connect()` action:
```js
connect() {
  this.swiper = new Swiper(this.element, {
    modules: [Navigation, Pagination],
    loop: true,
    navigation: {
      nextEl: '.swiper-button-next',
      prevEl: '.swiper-button-prev'
    },
    pagination: {
      el: '.swiper-pagination'
    }
    // other options as required...
  }
}
```
After changing it up this way, the Swiper element works as expected. The trick was that because I am using ESM modules with importmaps, I had to import the modules per https://swiperjs.com/swiper-api#using-js-modules

**Comment by mbajur** [comment]
I think i might be facing the original issue. I recently moved my app from esbuild to importmaps and my app is apparently not able to find some modules required by swiper. I pinned swiper and basically all of what @crespire described yet I'm getting couple of these:

<img width="1190" alt="image" src="https://github.com/user-attachments/assets/cc4af644-2101-4f52-bbf5-99769668f45d">

And ofcourse my carousel is not working. This should get reopened.
